### PR TITLE
Create optics_power_and_bias_current_test.go

### PR DESCRIPTION
**Pull request by sezhang2**
Validate optics input power, output power and bias current.

### CODE REVIEW

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	for _, tc := range cases {
+		t.Log(tc.desc)
+		intUpdateTime := 2 * time.Minute
+		t.Run(tc.desc, func(t *testing.T) {
+			i.Enabled = ygot.Bool(tc.IntfStatus)
+			dut.Config().Interface(dp.Name()).Replace(t, i)
+			dut.Telemetry().Interface(dp.Name()).OperStatus().Await(t, intUpdateTime, tc.expectedStatus)
+
+			// TODO: Remove t.Skipf() after the issue is fixed.
+			t.Skipf("Optics from INNOLIGHT is not supported, skip it for now.")
```
Comment (liulk) [inline feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:135]: Can we make this a conditional skip that detects the manufacturer?

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		})
+	}
+}
+
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
+	components := dut.Telemetry().ComponentAny().Name().Get(t)
+	var s []string
+	for _, c := range components {
+		lookupType := dut.Telemetry().Component(c).Type().Lookup(t)
+		if !lookupType.IsPresent() {
```
Comment (liulk) [inline feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:162]: Prefer to reduce the levels of nesting by eschewing else.

```
for _, c := range components {
  ...
  if !lookupType.IsPresent() {
    ...
    continue
  }
  componentType := lookupType.Val(t)
  ...
}
```

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+			t.Logf("Component %s type is not found", c)
```
Comment (liulk) [inline feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:163]: not found -> missing from telemetry?

Not found makes it sound like this function couldn't find the cType, but that's not really the case.

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+			componentType := lookupType.Val(t)
+			t.Logf("Component %s has type: %v", c, componentType)
+
+			switch v := componentType.(type) {
+			case telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT:
+				if v == cType {
+					s = append(s, c)
+				}
+			default:
+				t.Fatalf("Expected component type to be a hardware component, got (%T, %v)", componentType, componentType)
```
Comment (liulk) [inline feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:174]: Why do we fatal if there are non-hardware components? Software component is also allowed by OpenConfig. How about just ignore the component?

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	t.Logf("Found transceiver list: %v", transceivers)
+	if len(transceivers) == 0 {
+		t.Fatalf("Get transceiver list for %q: got 0, want > 0", dut.Model())
+	}
+
+	for _, transceiver := range transceivers {
+		t.Logf("Validate transceiver: %s", transceiver)
+		component := dut.Telemetry().Component(transceiver)
+
+		if !component.MfgName().Lookup(t).IsPresent() {
```
Comment (liulk) [inline feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:63]: Prefer to reduce the levels of nesting by eschewing else.

```
for _, transceiver := range transceivers {
  ...
  if !component.MgfName().Lookup(t).IsPresent() {
    t.Logf(...)
    continue
  }
  mfgName := ...
  ...
}
```

**Review by liulk — CHANGES_REQUESTED** [review]

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 5 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:63]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 2 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:170]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 4 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 5 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:63]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 3 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 2 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:170]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go (hunk 1 above)
Comment (sezhang2) [reply feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go]: done

File: feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go [hunk 6]
```diff
@@ -62,31 +62,31 @@ func TestOpticsPowerBiasCurrent(t *testing.T) {
...
-			t.Logf("Transceiver %s inputPowers: %v", transceiver, inputPowers)
-			if len(inputPowers) == 0 {
-				t.Errorf("Get inputPowers list for %q: got 0, want > 0", transceiver)
-			}
-			outputPowers := component.Transceiver().ChannelAny().OutputPower().Instant().Get(t)
-			t.Logf("Transceiver %s outputPowers: %v", transceiver, outputPowers)
-			if len(outputPowers) == 0 {
-				t.Errorf("Get outputPowers list for %q: got 0, want > 0", transceiver)
-			}
+		//TODO: Remove the mfgName INNOLIGHT check after the issue is fixed.
```
Comment (liulk) [inline feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go:70]:
```suggestion
		// TODO: Remove the mfgName INNOLIGHT check after the issue is fixed.
```

**Review by liulk — DISMISSED** [review]
I'm taking the next two days off, back on Monday.

**Review by liulk — APPROVED** [review]

**Pull request merged**